src/proto/uprotocol/upayload.rs

```diff
@@ -14,10 +14,66 @@
 use prost::Message;
 use prost_types::Any;
 
-use crate::{
-    cloudevent::serializer::SerializationError,
-    uprotocol::{Data, UPayload, UPayloadFormat},
-};
+pub use crate::types::serializationerror::SerializationError;
+use crate::uprotocol::{Data, UPayload, UPayloadFormat};
+
+const MIME_SUBSTYPE_JSON: &str = "json";
+const MIME_SUBTYPE_PROTOBUF: &str = "x-protobuf";
+const MIME_SUBTYPE_RAW: &str = "octet-stream";
+const MIME_SUBTYPE_SOMEIP: &str = "x-someip";
+const MIME_SUBTYPE_SOMEIP_TLV: &str = "x-someip_tlv";
+const MIME_SUBTYPE_PLAIN: &str = "plain";
+
+impl UPayloadFormat {
+    /// Gets the payload format that corresponds to a given MIME type.
+    ///
+    /// # Returns
+    ///
+    /// The corresponding payload format or [`UPayloadFormat::UpayloadFormatProtobuf`] if the MIME
+    /// type is unknown or empty.
+    pub fn from_mime_type(mime_type: &str) -> Self {
+        if let Ok(mime) = mime_type.parse::<mime::Mime>() {
+            match (mime.type_(), mime.subtype().as_str()) {
```

Comment:
From what I can see the `mime` crate exposes some of the common subtypes, but doesn't appear to expose all of the ones of interest to us (e.g. `x-protobuf` is missing).

I'd like it a bit more if they did and we could rely on using e.g. `mime::X_PROTOBUF` directly.

I think this is fine tho.

Wondering if we should try to upstream more of these subtypes we're using. WDYT?

Reply:
In general, I share your concern. However, `application/x-protobuf` is **not** an official (i.e. registered) MIME type ...